packetinjector/server_fake.go

```diff
@@ -0,0 +1,58 @@
+// +build !packetinject
+
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package packetinjector
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/ondemand"
+
+	"github.com/skydive-project/skydive/api/types"
+	"github.com/skydive-project/skydive/graffiti/graph"
+)
+
+type onDemandPacketInjectServer struct {
+	sync.RWMutex
+	graph *graph.Graph
+}
+
+func (o *onDemandPacketInjectServer) CreateTask(srcNode *graph.Node, resource types.Resource) (ondemand.Task, error) {
+	logging.GetLogger().Debugf("Ingonring packet injection %s on %s because it was disabled at compile time with \"nopacketinject\"", resource.ID(), srcNode.ID)
```

Comment:
I would return an error here instead of logging. It will be logged any way
```
return nil, fmt.Errorf(...)
```

Reply:
Done